### DISCUSSION TITLE
How to add more generic-uio devices to icicle dev kit

### DISCUSSION THREAD

**Discussion by es-ggg** [polarfire-soc/polarfire-soc-documentation#390] [How to] [answered]
Hello,

I'm attempting to modify the icicle dev kit device trees to add at least 4 more generic-uio devices to the /dev directory and also know the addresses at which I can access these newly added devices. By default the icicle dev kit comes with 3 uio devices uio0, uio1 and uio2. I want to add more nodes from uio3 to uio6 and I don't know how to do this. 

In the **~/yocto-dev/build/tmp-glibc/work-shared/icicle-kit-es/kernel-source/arch/riscv/boot/dts/microchip/mpfs-icicle-kit-fabric.dtsi**, the following node exists:

fpgalsram: uio@60000000 {
                        compatible = "generic-uio";
                        linux,uio-name = "fpga_lsram";
                        reg = <0x0 0x60000000 0x0 0x1000>;
                        status = "disabled";
                };

I don't know how this node corresponds to the 3 uio devices in the /dev directory: uio0, uio1 and uio2, and what the addresses to access the uio1 and uio2 nodes would be using devmem2. I also don't know how to modify any of the existing device trees to add more uio nodes at an available address on the fabric. Any advice, insight or documentation would be greatly appreciated.

Thanks in advance,
SG

**Comment by eppidei** [comment, marked as answer]
Hi @es-ggg ,

to enable a new node you just need to add similar additional entries in the device tree.
It follows the usual device tree syntax rules, As for the mapping of the /dev/uioX to a specific peripheral you should find the details in the /sys file systems.

**Reply by es-ggg** [reply]
Thank you very much @eppidei! That worked.
